pkg/endpoint/policy.go

```diff
@@ -923,6 +931,85 @@ func (e *Endpoint) runIdentityToK8sPodSync() {
 	)
 }
 
+// FormatMetadata returns metdata about the endpoint in the format
+// / <global ID Prefix>:<cluster name>:<node name>:<endpoint ID> as a string.
+func (e *Endpoint) FormatMetadata() string {
+	var metadata []string
+	metadata = append(metadata, CiliumGlobalIdPrefix)
+	metadata = append(metadata, ipcache.AddressSpace)
+	nodeIdentity, _ := node.GetLocalNode()
+	metadata = append(metadata, nodeIdentity.Name)
+	metadata = append(metadata, strconv.Itoa(int(e.ID)))
+	return strings.Join(metadata, ":")
+
+}
+
+// This synchronizes the key-value store with a mapping of the endpoint's IP
+// with the numerical ID representing its security identity.
+func (e *Endpoint) runIPIdentitySync(endpointIP addressing.CiliumIP) {
+
+	if endpointIP == nil {
+		return
+	}
+
+	var addressFamily string
+	if endpointIP.IsIPv6() {
+		addressFamily = "ipv6"
+	} else {
+		addressFamily = "ipv4"
+	}
+
+	e.controllers.UpdateController(fmt.Sprintf("sync-%s-identity-mapping (%d)", addressFamily, e.ID),
+		controller.ControllerParams{
+			DoFunc: func() error {
+
+				ipKey := path.Join(ipcache.IPIdentitiesPath, ipcache.AddressSpace, endpointIP.String())
```

Comment:
You can move this to outside of the controller so you don't have to recalculate it in the `StopFunc()`